From: Marcus Watts <mwatts@redhat.com>
Date: Thu, 3 Nov 2016 23:33:44 -0400
Subject: rgw: cmake: remove useless civetweb include path side effect.

For 'target_include_directories" for the cmake object library
'civetweb_common_objs', change from PUBLIC to PRIVATE.  This doesn't
break anything, so it wasn't doing anything useful.  If it has it
any effect, it would be to cause everything that linked against this
"library" to also use the indictated include path.  Which would be great
except everything in ceph wants to include "civetweb/civetweb.h" and
not "civetweb.h".  We already make separate arrangements elsewhere for
that to work.  Additionally, static object libraries in cmake aren't
really libraries, so I'm not entirely sure this even does anything.
So: making this public is not useful, and could be harmful.  Making it
private makes this only take effect for building civetweb.c itself,
exactly the effect we we require, and no more.

Signed-off-by: Marcus Watts <mwatts@redhat.com>
(cherry picked from commit 8308a13b0257c9460fd2a721c20b0c37cb9e7c57)
(cherry picked from commit 386640865dee30d38f17e55fc87535e419bc3cb5)

Resolves: rhbz#1258961
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ca03194..f51ad81 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1175,7 +1175,7 @@ if(${WITH_RADOSGW})
 
   set(civetweb_common_files civetweb/src/civetweb.c)
   add_library(civetweb_common_objs OBJECT ${civetweb_common_files})
-  target_include_directories(civetweb_common_objs PUBLIC
+  target_include_directories(civetweb_common_objs PRIVATE
 	"${CMAKE_SOURCE_DIR}/src/civetweb/include")
   if(HAVE_OPENSSL)
   set_property(TARGET civetweb_common_objs
